.github/workflows/author_approval.yml

```diff
@@ -0,0 +1,42 @@
+name: Author Approval Label
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  label:
+    permissions:
+        pull-requests: write
+    runs-on: ubuntu-latest
+    # Run on comments that are:
+    # 1) on PRs, not issues,
+    # 2) not from bot users and,
+    # 3) include the string "approved"
+    # If so, we will do the work to check that the commenter is the package author,
+    # and conditionally apply the author-approved label.
+    if: ${{ github.event.issue.pull_request && github.event.issue.user.type != 'Bot' && contains(github.event.comment.body, 'approved') }}
```

Comment:
The word `approved` will need to appear with exact case matching (case sensitive) in the document, right? Should we document this somewhere?

Reply:
Yep, I think AutoMerge should mention it any time there’s a failure due to lack of approval (none exist yet). Perhaps once we have some we can add a faq entry as well. I don’t think it makes sense to add public docs with this PR as the mechanism isn’t used for anything yet.

Reply:
For now can you stick a comment in the YAML (right above this line, maybe) just pointing out that it's case sensitive?

Reply:
```suggestion
    # note: `approved` here is case-sensitive 
    if: ${{ github.event.issue.pull_request && github.event.issue.user.type != 'Bot' && contains(github.event.comment.body, 'approved') }}
```